src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -437,22 +504,43 @@ private async Task<bool> TryReadPrefaceAsync()
         return false;
     }
 
-    private static bool ParsePreface(in ReadOnlySequence<byte> buffer, out SequencePosition consumed, out SequencePosition examined)
+    private bool ParseHttp1x(ReadOnlySequence<byte> buffer, out HttpVersion httpVersion)
     {
-        consumed = buffer.Start;
-        examined = buffer.End;
+        httpVersion = HttpVersion.Unknown;
 
-        if (buffer.Length < ClientPreface.Length)
+        var reader = new SequenceReader<byte>(buffer);
```

Comment:
We should probably trim this to at most `Limits.MaxRequestLineSize` before calling `TryReadTo` similar to what we do here:

https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs#L179-L204

Reply:
Done